test/gtest/ucs/test_datatype.cc

```diff
@@ -523,3 +537,205 @@ UCS_TEST_F(test_datatype, ptr_status) {
     void *ptr2 = (void*)(uintptr_t)(UCS_ERR_LAST + 1);
     EXPECT_TRUE(UCS_PTR_IS_ERR(ptr2));
 }
+
```

Comment:
I think we don't need to duplicate the correctness checks - just the performance checks...

Reply:
Well, there's no other test for the locked ptr_array.
I recommend we keep the new tests (replicas of the ptr_array tests).